hack/helm/templates/hnc-manager.yaml

```diff
@@ -0,0 +1,721 @@
+apiVersion: v1
```

Comment:
Instead of just checking this in by hand, can you please update the makefile to regenerate it? It's ok to check this in (I think Helm _requires_ that it's checked in, right?), but other than committing it, a developer shouldn't have to do something other than (say) `make manifests`. Otherwise this will quickly drift away from the regular version.

Also, can you please add a comment at the top of this file (automatically inserted, of course) saying something like "Do not edit this file by hand; run 'make manifests' [or whatever] to update."

Thanks!

Reply:
that's one of the todo items / why it's a draft PR. Wanted some initial feedback before proceeding down the wrong routes :)

Reply:
So we should discuss what to generate and how.

1. Helm standards / common practices for example is to split a file like this up such that there's a `deployment.yaml` file a `serviceaccount.yaml` file and so on. Is it feasible to do this / do we want this?

2. Helm provides a mechanism for namespace creation and targetting. So in theory namespace creation shouldn't be part of the helm chart and anywhere there's a namespace defined it should be a variable `namespace: {{ .Release.Namespace }}`

3. Helm provides a mechanism for label definition again how challenging will this be technically / how badly do we want or need it ie
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: {{ include "temp.fullname" . }}
  labels:
    {{- include "temp.labels" . | nindent 4 }}
spec:
  {{- if not .Values.autoscaling.enabled }}
  replicas: {{ .Values.replicaCount }}
  {{- end }}
  selector:
    matchLabels:
      {{- include "temp.selectorLabels" . | nindent 6 }}
```

I'm would prefer incremental improvement so a first pass of "an entire static yaml file is generated" with backlog tickets for further helm-ification would be my personal preference but worth discussing before assuming that route is suitable.

Reply:
Thanks! And yes, I'm just giving early feedback, I realize this is a draft.

I'd personally prioritize "ease of automation" over "following all the Helm guidelines" for now. I'd rather have something up-to-date and a little nonstandard than a perfect but obsolete chart.

For `deployment.yaml` vs `serviceaccount.yaml` - I think I'd say that if you can generate them from anything in `config/` or from the output of kustomize, then go for it. We can also iterate on that - start as a monolithic file and then use something like the Gatekeeper tooling to split it up.

re namespace - I don't _think_ we use the string `hnc-system` anywhere else, so you could totally add a `sed` command in the makefile to replace it with `{{ .Release.Namespace }}`. Note that there might be strings that _include_ `hnc-system` but that should be pretty easy to check and/or exclude. But I'm totally ok with _not_ doing any of that as well. I think the number of people who would object to `hnc-system` as well as the other names and labels inside that namespace should be vanishingly small, and I'd be very happy with _never_ adding that flexibility unless someone asks for it and/or is willing to do it themselves.